holoviews/plotting/mpl/plot.py

```diff
@@ -82,8 +82,7 @@ class MPLPlot(DimensionedPlot):
     sublabel_size = param.Number(default=18, doc="""
          Size of optional subfigure label.""")
 
-    projection = param.ObjectSelector(default=None,
-                                      objects=['3d', 'polar', None], doc="""
+    projection = param.Parameter(default=None, doc="""
```

Comment:
Do you validate the possible strings? I've not read the code below but we should make sure if a string is supplied it is validate...